src/Workspaces/MSBuildTest/OpenProjectsTests.cs

```diff
@@ -0,0 +1,225 @@
+        }
+
+        [ConditionalTheory(typeof(DotNetSdkMSBuildInstalled))]
+        [MemberData(nameof(GetVisualBasicProjectTemplateNames), DisableDiscoveryEnumeration = false)]
+        public async Task ValidateVisualBasicTemplateProjects(string templateName)
+        {
+            await AssertTemplateProjectLoadsCleanlyAsync(templateName, LanguageNames.VisualBasic);
+        }
+
+        public static TheoryData<string> GetCSharpProjectTemplateNames()
+            => GetProjectTemplateNames("c#");
+
+        public static TheoryData<string> GetVisualBasicProjectTemplateNames()
+            => GetProjectTemplateNames("vb");
+
+        public static TheoryData<string> GetProjectTemplateNames(string language)
+        {
+            // The expected output from the list command is as follows.
+
+            // These templates matched your input: --language='vb', --type='project'
+            //
+            // Template Name                  Short Name           Language  Tags
+            // -----------------------------  -------------------  --------  ---------------
+            // Class Library                  classlib             C#,F#,VB  Common/Library
+            // Console App                    console              C#,F#,VB  Common/Console
+            // ...
+
+            var result = RunDotNet($"new list --type project --language {language}");
+            Assert.Equal(0, result.ExitCode);
+
+            var lines = result.Output
+                .Replace("\r\n", "\n")
+                .Split(new[] { "\n" }, StringSplitOptions.RemoveEmptyEntries);
+
+            TheoryData<string> templateNames = [];
+            var foundDivider = false;
+
+            foreach (var line in lines)
+            {
+                if (!foundDivider)
+                {
+                    if (line.StartsWith("----"))
+                    {
+                        foundDivider = true;
+                    }
+                    continue;
+                }
+
+                var columns = line.Split(new[] { "  " }, StringSplitOptions.RemoveEmptyEntries)
+                    .Select(c => c.Trim())
+                    .ToArray();
+                var templateShortName = columns[1].Split(',').First();
+
+                if (ExcludeMauiTemplates && templateShortName.StartsWith("maui"))
+                    continue;
+
+                templateNames.Add(templateShortName);
+            }
+
+            return templateNames;
+        }
+
+        private async Task AssertTemplateProjectLoadsCleanlyAsync(string templateName, string languageName, string[]? ignoredDiagnostics = null)
+        {
+            if (ignoredDiagnostics?.Length > 0)
+            {
+                TestOutputHelper.WriteLine($"Ignoring compiler diagnostics: \"{string.Join("\", \"", ignoredDiagnostics)}\"");
+            }
+
+            using var projectDirectory = (DisposableDirectory)SolutionDirectory;
```

Comment:
This cast seems a bit funky -- whoever created this should be disposing it, right? not us?

Reply:
OK, so I stopped digging when it looked like WorkspaceTestBase didn't bother disposing SolutionDirectory. Now I tracked down that the SolutionDirectory is disposed by TempRoot which gets disposed by TestBase when it is disposed. I also learned xUnit create a new test class instance for each test run. So yes this is unnecessary. It felt icky to write. Thanks for having me follow it further.